modules/programs/contour.nix

```diff
@@ -0,0 +1,1193 @@
+              Numpad_0, Numpad_1, Numpad_2, Numpad_3, Numpad_4,
+              Numpad_5, Numpad_6, Numpad_7, Numpad_8, Numpad_9.
+
+              In case of standard characters, just the character.
+            '';
+          };
+
+          mouse = mkOption {
+            type = types.nullOr
+              (types.enum [ "Left" "Middle" "Right" "WheelUp" "WheelDown" ]);
+            default = null;
+            description = ''
+              Mouse button specification.
+            '';
+          };
+
+          action = mkOption {
+            type = types.enum [
+              "CancelSelection"
+              "ChangeProfile"
+              "ClearHistoryAndReset"
+              "CopyPreviousMarkRange"
+              "CopySelection"
+              "DecreaseFontSize"
+              "DecreaseOpacity"
+              "FollowHyperlink"
+              "IncreaseFontSize"
+              "IncreaseOpacity"
+              "NewTerminal"
+              "OpenConfiguration"
+              "OpenFileManager"
+              "PasteClipboard"
+              "PasteSelection"
+              "Quit"
+              "ReloadConfig"
+              "ResetConfig"
+              "ResetFontSize"
+              "ScreenshotVT"
+              "ScrollDown"
+              "ScrollMarkDown"
+              "ScrollMarkUp"
+              "ScrollOneDown"
+              "ScrollOneUp"
+              "ScrollPageDown"
+              "ScrollPageUp"
+              "ScrollToBottom"
+              "ScrollToTop"
+              "ScrollUp"
+              "SendChars"
+              "ToggleAllKeyMaps"
+              "ToggleFullScreen"
+              "ToggleTitleBar"
+              "ViNormalMode"
+              "WriteScreen"
+            ];
+            description = ''
+              Action to be executed on mapped input activation.
+            '';
+          };
+
+          mode = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = lib.mdDoc ''
+              Additionally one can filter input mappings based on special terminal modes using the `modes` option:
+
+              * Alt       : The terminal is currently in alternate screen buffer, otherwise it is in primary screen buffer.
+              * AppCursor : The application key cursor mode is enabled (otherwise it's normal cursor mode).
+              * AppKeypad : The application keypad mode is enabled (otherwise it's the numeric keypad mode).
+              * Select    : The terminal has currently an active grid cell selection (such as selected text).
+              * Insert    : The Insert input mode is active, that is the default and one way to test that the input mode is not in normal mode or any of the visual select modes.
+
+              You can combine these modes by concatenating them via | and negate a single one
+              by prefixing with ~.
+            '';
+          };
+        };
+      });
+      default = [
```

Comment:
Can't the default simply be `[ ]`?

Reply:
That would result in Contour having no keybindings at all, which is rather unexpected and requires user to re-generate config manually to obtain default values

Reply:
Ah, that's unfortunately. I imagine you will then have to do something like in the i3 module: https://github.com/nix-community/home-manager/blob/1a8e35d2e53ed2ccd9818fad9c9478d56c655661/modules/services/window-managers/i3-sway/i3.nix#L22-L111

Note, the example and description shows the user how they can extend the default keybindings.